src/Microsoft.AspNetCore.OData/Results/BadRequestODataResult.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Results
+{
+    /// <summary>
+    /// Represents a result that when executed will produce a Bad Request (400) response.
+    /// </summary>
+    /// <remarks>This result creates an <see cref="ODataError"/> with status code: 400.</remarks>
+    public class BadRequestODataResult : BadRequestResult, IODataErrorResult
+    {
+        /// <summary>
+        /// OData Error.
+        /// </summary>
+        public ODataError Error { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the class.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        public BadRequestODataResult(string message)
+        {
+            Error = new ODataError
+            {
+                Message = message,
+                ErrorCode = "400"
+            };
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the class.
+        /// </summary>
+        /// <param name="odataError">OData Error.</param>
+        public BadRequestODataResult(ODataError odataError)
+        {
+            Error = odataError;
```

Comment:
>Error = odataError; [](start = 12, length = 19)

check the null for the input odataError

Reply:
same for other?

---
In reply to: [541149484](https://github.com/OData/WebApi/pull/2341#discussion_r541149484) [](ancestors = 541149484)

Reply:
#Resolved